### PR TITLE
Use OpenAPI Power Gem

### PR DESCRIPTION
**Pull request by agrare**
Use the ibm_cloud_power OpenAPI gem

TODO:
- [x] Update connection methods (raw_connect, verify_credentials, etc..)
- [x] Update Refresh
- [x] Update operations
- [x] Attach/Detach volumes

Depends:
- [x] https://github.com/IBM-Cloud/ibm-cloud-sdk-ruby/pull/8
- [x] Release ibm_cloud_iam
- [x] Release ibm_cloud_resource_controller
- [x] Release ibm_cloud_power

### CODE REVIEW

File: app/models/manageiq/providers/ibm_cloud/power_virtual_servers/storage_manager/cloud_volume.rb [hunk 1]
```diff
@@ -152,8 +156,8 @@ def validate_attach_volume
   end
 
   def raw_attach_volume(vm_ems_ref, _device = nil)
-    ext_management_system.with_provider_connection(:service => 'PowerIaas') do |power_iaas|
-      power_iaas.attach_volume(vm_ems_ref, ems_ref)
+    with_provider_connection(:service => 'PCloudPVMInstancesApi') do |api|
+      # TODO
```
Comment (agrare) [inline app/models/manageiq/providers/ibm_cloud/power_virtual_servers/storage_manager/cloud_volume.rb:160]: TODO I didn't see a pvm_instances_volume_attach API call

File: app/models/manageiq/providers/ibm_cloud/power_virtual_servers/storage_manager/cloud_volume.rb [hunk 2]
```diff
@@ -165,8 +169,8 @@ def validate_detach_volume
   end
 
   def raw_detach_volume(vm_ems_ref)
-    ext_management_system.with_provider_connection(:service => 'PowerIaas') do |power_iaas|
-      power_iaas.detach_volume(vm_ems_ref, ems_ref)
+    with_provider_connection(:service => 'PCloudPVMInstancesApi') do |api|
+      # TODO
```
Comment (agrare) [inline app/models/manageiq/providers/ibm_cloud/power_virtual_servers/storage_manager/cloud_volume.rb:173]: Same as https://github.com/ManageIQ/manageiq-providers-ibm_cloud/pull/56/files#r523197943

File: app/models/manageiq/providers/ibm_cloud/power_virtual_servers/storage_manager/cloud_volume.rb (hunk 1 above)
Comment (agrare) [reply app/models/manageiq/providers/ibm_cloud/power_virtual_servers/storage_manager/cloud_volume.rb]: `cloud_pvminstances_volumes_post`/`cloud_pvminstances_volumes_delete`

File: manageiq-providers-ibm_cloud.gemspec [hunk 3]
```diff
@@ -19,6 +19,9 @@ Gem::Specification.new do |spec|
   spec.require_paths = ["lib"]
 
   spec.add_dependency "ibm-cloud-sdk", "~> 0.1"
+  spec.add_dependency "ibm_cloud_iam", "~> 1.0"
+  spec.add_dependency "ibm_cloud_power", "~> 1.0"
+  spec.add_dependency "ibm_cloud_resource_controller", "~> 1.0"
```
Comment (Fryguy) [inline manageiq-providers-ibm_cloud.gemspec:24]: alphabetical?

**Comment by Fryguy** [comment]
Does this depend on https://github.com/IBM-Cloud/ibm-cloud-sdk-ruby/pull/8 (and subsequent release?)

File: manageiq-providers-ibm_cloud.gemspec (hunk 3 above)
Comment (agrare) [reply manageiq-providers-ibm_cloud.gemspec:23]: :+1:

**Comment by agrare** [comment]
> Does this depend on IBM-Cloud/ibm-cloud-sdk-ruby#8 (and subsequent release?)

Yes it does, I'm going to drop `vpc` from that since it is going to require some more work but iam, resource_controller, and power are all ready

**Comment by Fryguy** [comment]
Can you update the OP to show that dependency?

**Comment by Fryguy** [comment]
@agrare Is this ready to go now?

**Comment by agrare** [comment]
@Fryguy yes I just released the gems and kicked the tests

**Review by Fryguy — APPROVED** [review]
@jaywcarman Can you do the honors?

**Pull request merged**